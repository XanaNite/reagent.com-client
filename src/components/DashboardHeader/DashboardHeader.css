.dashboardHeader{
    padding: 10px;
}

.hamburgerMenu{
    size: 30px;
    margin-right: 15px;
}

.dashboard-title h1{
    margin: 0;
}

.dashboardImg-container{
    font-size: 30px;
    margin: 5px;
}

@media screen and (min-width: 490px){
    .dashboard-title{
        display: flex;
    }

    .dashboard-title span{
        margin-left: 10px;
    }
}