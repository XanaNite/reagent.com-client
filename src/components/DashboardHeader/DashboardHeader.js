import React from 'react';
import {Link} from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser } from '@fortawesome/free-solid-svg-icons';
import './DashboardHeader.css';
import HamburgerMenu from '../HamburgerMenu/HamburgerMenu';

export default function DashboardHeader(){
    return(
        <header role="banner" className='dashboardHeader'>
            <nav role="navigation" className="flexbox-between">
                <div className="hamburgerMenu">
                    <HamburgerMenu />
                </div>
                <div className="dashboard-title">
                    <h1><Link to='/'>REagent.com</Link></h1>
                    <span>for Professionals</span>
                </div>
                <div className="dashboardImg-container">
                    <Link to='/myprofile'><FontAwesomeIcon icon={faUser} /></Link>
                </div>
            </nav>
        </header>
    );
}