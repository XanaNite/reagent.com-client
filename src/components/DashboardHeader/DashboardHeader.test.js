import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DashboardHeader from './DashboardHeader';

describe(`DashboardHeader component`, () => {
    const props = {
        className: 'test-class-name',
        children: <p>test children</p>,
        'data-other': 'test-other-prop',
        agent: {
            first_name: "test",
            last_name: "test",
            title: "test",
            city: "test",
            state: "test"
        }
    };

    it('renders the DashboardHeader given props', () => {
        const wrapper = shallow(<DashboardHeader {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });
})