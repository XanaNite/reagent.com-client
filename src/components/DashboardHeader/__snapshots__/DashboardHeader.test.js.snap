// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardHeader component renders the DashboardHeader given props 1`] = `
<header
  className="dashboardHeader"
  role="banner"
>
  <nav
    className="flexbox-between"
    role="navigation"
  >
    <div
      className="hamburgerMenu"
    >
      <Link
        to="/dashboard"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f0c9",
                "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
              ],
              "iconName": "bars",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
    <div
      className="dashboard-title"
    >
      <h1>
        <Link
          to="/"
        >
          REagent.com
        </Link>
      </h1>
      <span>
        for Professionals
      </span>
    </div>
    <div
      className="dashboardImg-container"
    >
      <Link
        to="/myprofile"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Object {
              "icon": Array [
                448,
                512,
                Array [],
                "f007",
                "M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
              ],
              "iconName": "user",
              "prefix": "fas",
            }
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </Link>
    </div>
  </nav>
</header>
`;
